src/main/java/com/example/dbdemo/model/Faculty.java

```diff
@@ -3,85 +3,61 @@
 import com.example.dbdemo.utilities.Gender;
 import jakarta.persistence.*;
 import jakarta.validation.constraints.*;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
+import java.io.Serializable;
 import java.time.LocalDate;
-import java.util.Date;
-import java.util.List;
 import java.util.Set;
 
 @Entity
 @Data
-@Table(name="faculty")
-public class Faculty {
+@Table(name = "faculty")
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+public class Faculty implements Serializable {
+
     @Id
     @GeneratedValue(strategy = GenerationType.SEQUENCE)
     private Long id;
 
-    @NotBlank(message = "Name should not be null or empty")
-    @Size(max = 255, message = "Name should not exceed 255 characters.")
-    @Pattern(regexp = "^[a-zA-Z ]*$", message = "Name should contain only letters.")
     private String name;
 
-    @NotBlank(message = "Father's name should not be null or empty")
-    @Size(max = 255, message = "Father's name should not exceed 255 characters.")
-    @Pattern(regexp = "^[a-zA-Z ]*$", message = "Father's name should contain only letters.")
     private String fatherName;
 
-    @NotBlank(message = "Mother's name should not be null or empty")
-    @Size(max = 255, message = "Mother's name should not exceed 255 characters.")
-    @Pattern(regexp = "^[a-zA-Z ]*$", message = "Mother's name should contain only letters.")
     private String motherName;
 
-    @NotBlank
-    @Size(min = 10, max = 10, message = "Contact number should be 10 digits.")
-    @Pattern(regexp = "^[0-9]*$", message = "Contact number should not contain special characters.")
     private String contactNumber;
 
-    @Size(min = 10, max = 10, message = "Backup contact number should be 10 digits.")
-    @Pattern(regexp = "^[0-9]*$", message = "Backup contact number should not contain special characters.")
     private String backupContactNumber;
 
-    @NotNull
-    @Past(message = "Date of birth should be in the past.")
-    @Temporal(TemporalType.DATE)
     private LocalDate dateOfBirth;
 
-    @NotBlank
-    @Email(message = "Invalid email format.")
     private String email;
 
-    @NotNull
     @Enumerated(EnumType.STRING)
     private Gender gender;
 
-    @Size(min = 12, max = 12, message = "Aadhar number should be 12 digits.")
-    @Pattern(regexp = "^[0-9]*$", message = "Aadhar number should not contain special characters.")
     private String aadharNumber;
 
-    @NotBlank
-    @Pattern(regexp = "^[a-zA-Z0-9]*$", message = "PAN should be alphanumeric with no special characters.")
-    @Size(min = 10, max = 10, message = "PAN should be 10 characters.")
     private String pan;
 
     private String maritalStatus;
 
     @PastOrPresent(message = "Date of joining should be in the past or present.")
+    @Builder.Default
     private LocalDate dateOfJoining = LocalDate.now();
```

Comment:
pls check the format only need the date part, not all of yyyy-mm-ddThh:mm:sec.microsec
you can use Date also